services/src/handlers/datasets.rs

```diff
@@ -370,58 +396,56 @@ fn auto_detect_meta_data_definition(main_file_path: &Path) -> Result<MetaDataDef
 
     let time = detect_time_type(&columns_vecs);
 
-    Ok(MetaDataDefinition::OgrMetaData(StaticMetaData::<
-        _,
-        _,
-        VectorQueryRectangle,
-    > {
-        loading_info: OgrSourceDataset {
-            file_name: main_file_path.into(),
-            layer_name: geometry.layer_name.unwrap_or_else(|| layer.name()),
-            data_type: Some(geometry.data_type),
-            time,
-            default_geometry: None,
-            columns: Some(OgrSourceColumnSpec {
-                format_specifics: None,
-                x,
-                y,
-                int: columns_vecs.int,
-                float: columns_vecs.float,
-                text: columns_vecs.text,
-                bool: vec![],
-                datetime: columns_vecs.date,
-                rename: None,
-            }),
-            force_ogr_time_filter: false,
-            force_ogr_spatial_filter: false,
-            on_error: geoengine_operators::source::OgrSourceErrorSpec::Ignore,
-            sql_query: None,
-            attribute_query: None,
-        },
-        result_descriptor: VectorResultDescriptor {
-            data_type: geometry.data_type,
-            spatial_reference: geometry.spatial_reference,
-            columns: columns_map
-                .into_iter()
-                .filter_map(|(k, v)| {
-                    v.try_into()
-                        .map(|v| {
-                            (
-                                k,
-                                VectorColumnInfo {
-                                    data_type: v,
-                                    measurement: Measurement::Unitless,
-                                },
-                            )
-                        })
-                        .ok()
-                }) // ignore all columns here that don't have a corresponding type in our collections
-                .collect(),
-            time: None,
-            bbox: None,
+    Ok(crate::datasets::storage::MetaDataDefinition::OgrMetaData(
```

Comment:
shouldn't this be the OgrMetaData from services/api/model/servces ?

Reply:
I thought of `auto_detect_meta_data_definition` more of like an internal utility function which isn't directly connected to the API. Because of that I used internal types.